### PR TITLE
🎖️ Bookstore: connect to API 🎖️

### PR DESCRIPTION
**Pull request by kambereBr**
Project Requirements : https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connect_to_api_v2_1.md

## 📝 General requirements

- ✅ No Linters error
- ✅ Used GitFlow.
- ✅ Understood the concept of state management.
- ✅ Used store, actions, and reducers in React.
- ✅ Connected React and Redux.
- ✅ Connected  [(Bookstore-API)](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) via Redux and used the data to display something in a UI.
- ✅ Refactored **add book** and **remove book** features to persist changes in the server
- ✅ The application runs in the browser without errors

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,32 +1,60 @@
...
+      }, {
+        headers: {
+          'Content-type': 'application/json; charset=UTF-8',
+        },
+      });
+      return { message: resp, id };
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Oooops, something went wrong');
+    }
+  });
```
Comment (taldr27) [inline src/redux/books/booksSlice.js]: - [ ] I noticed a little issue with your deletebook function in the video I've shared below. The deletion process works fine, but it seems that the UI doesn't update accordingly. Additionally, if I click the delete button again, the book disappears visually, even though in the network tab it's evident that the book was already deleted. Could you please take a look and ensure that your UI updates as expected?

- [ ] I'd also like to mention a quirky error message I spotted in the console while trying to add and remove books. It's possible that this error might be related to the deletion problem mentioned earlier. Would you mind checking into this? 


   Here's the video for a better understanding.
   
   [scrnli_24_8_2023_18-45-27.webm](https://github.com/kambereBr/bookstore/assets/107952724/fb4193b0-63d4-4810-a5b2-fb9532651558)

**Review by taldr27 — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️
Hello @kambereBr, 

Good job so far! 
You're making good progress on the project, but some issues still need to be addressed before you can move on to the next stage. Keep up the good work! 💪

almost-there

## Highlights 🎉
✅ Professional Readme. 
✅ Linters passing.
✅ Correct use of GitHub Flow.
✅ Good job creating your asyncthunks to add and remove your books.
✅ Good job using axios in your project.

## Required Changes ♻️
_Check the comments under the review._

------
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @taldr27 in your question so I can receive the notification.

If you need any further help or have any questions, feel free to reach out to me on Slack as _**Diego Garcia**_.🎯

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). gif_

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (kambereBr) [reply src/redux/books/booksSlice.js]: Thanks dear @taldr27. Now, All issues fixed 👏

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @kambereBr,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by kambereBr** [comment]
Thanks dear @Omar-Muhamad 👏

**Pull request merged**